Rakefile

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require 'date'
+require 'json'
+require 'net/http'
+require 'optparse'
+
+require 'curb' # Curl Ruby, an external libcurl binding library
+
+UNDESIRED_LABELS = [
+  'refactoring', 'needs conceptual review', 'needs rebase', 'waiting for author'
+]
+GITHUB_API_URL = 'https://api.github.com/repos/bitcoin/bitcoin/pulls'
+HTTP_SUCCESS = 200
+HTTP_ERRORS = [
+  JSON::ParserError,
+  EOFError,
+  Errno::ECONNRESET,
+  Errno::EINVAL,
+  Net::HTTPBadResponse,
+  Net::HTTPHeaderSyntaxError,
+  Net::ProtocolError,
+  Timeout::Error
+]
+
+desc 'Create a new post file'
+namespace :posts do
+  task :new do
+    # Fetch user command line arguments.
+    pr, host, date = get_cli_options
+
+    # Set default value for meeting date if not supplied by user.
+    date ||= next_wednesday.to_s
+
+    # Ensure date is valid.
+    unless valid_iso8601_date?(date)
+      puts "Error: Invalid date (#{date} received, YYYY-MM-DD needed). Nothing done, exiting."
+      exit
+    end
+
+    # Fetch pull request data from the GitHub v3 REST API.
+    http = Curl.get("#{GITHUB_API_URL}/#{pr}") { |request| request.headers = headers }
+    response = parse_response(http)
+
+    # Create a new post file if none exists.
+    filename = "#{'_posts/' if File.directory?('_posts')}#{date}-##{pr}.md"
+    if File.file?(filename)
+      puts "Error: Filename #{filename} already exists! Nothing done, exiting."
+      exit
+    else
+      create_post_file!(filename, response, date, host)
+      puts "New file #{filename} created successfully."
+      exit
+    end
+  end
+end
+
+def get_cli_options(options = {})
+  OptionParser.new do |opts|
+    opts.banner = 'Usage: rake posts:new -- <options>'
+    opts.on('--pr NUMBER', 'Pull request number') { |pr| options[:pr] = pr }
```

Comment:
Is it possible to make this argument required and exit with a helpful error and print out the help text.

Currently, the error is not very helpful:

```
→ rake posts:new
HTTP 404 Not Found
{"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
```

Reply:
Agreed, was about to push this:

"Error: Missing --pr argument (required). Run `rake posts:new -- --help` for help info."